aspnet/fundamentals/file-providers.rst

```diff
@@ -1,8 +1,154 @@
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/project.json
+  :lines: 42-49
+  :emphasize-lines: 4-7
+
+You can use `globbing patterns`_ when specifying files to embed in the assembly.
+
+.. note:: It's unlikely you would ever want to actually embed every .js file in your project in its assembly; the above sample is for demo purposes only.
+
+When creating an ``EmbeddedFileProvider``, pass the assembly it will read to its constructor.
+
+.. code-block:: c#
+
+  var embeddedProvider = new EmbeddedFileProvider(Assembly.GetEntryAssembly());
+
+The snippet above demonstrates how to create an ``EmbeddedFileProvider`` with access to the currently executing assembly.
+
+Updating the sample app to use an ``EmbeddedFileProvider`` results in the following output:
+
+.. image:: /fundamentals/file-providers/_static/embedded-directory-listing.png
+
+.. note:: Embedded resources do not expose directories. Rather, the path to the resource (via its namespace) is embedded in its filename using ``.`` separators.
+
+.. tip:: The ``EmbeddedFileProvider`` constructor accepts an optional ``baseNamespace`` parameter. Specifying this will scope calls to ``GetDirectoryContents`` to those resources under the provided namespace.
+
+CompositeFileProvider
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``CompositeFileProvider`` combines ``IFileProvider`` instances, exposing a single interface for working with files from multiple providers. When creating the ``CompositeFileProvider``, you pass one or more ``IFileProvider`` instances to its constructor:
+
+.. literalinclude:: file-providers/sample/src/FileProviderSample/Startup.cs
+  :lines: 35-37
+  :emphasize-lines: 3
+  :dedent: 12
+
+Updating the sample app to use a ``CompositeFileProvider`` that includes both the physical and embedded providers configured previously, results in the following output:
+
+.. image:: /fundamentals/file-providers/_static/composite-directory-listing.png
+
+
+Watching for changes
+--------------------
+
+The ``IFileProvider`` ``Watch`` method provides a way to watch one or more files or directories for changes. This method accepts a path string, which can use `globbing patterns`_ to specify multiple files, and returns an ``IChangeToken``. This token exposes a ``HasChanged`` property that can be inspected, and a ``RegisterChangeCallback`` method that is called when changes are detected to the specified path string. Note that each change token only calls its associated callback in response to a single change. To enable constant monitoring, you can use a ``TaskCompletionSource`` as shown below, or re-create ``IChangeToken`` instances in response to changes.
+
+In this article's sample, a console application is configured to display a message whenever a text file is modified:
+
+.. literalinclude:: file-providers/sample/src/WatchConsole/Program.cs
```

Comment:
I double-checked the style guide, and according to current guidelines this usage appears to count as one of the exceptions where it's OK, so it's up to you -- here's the rule:  "Don't use foo, foobar, or related words in content for a general audience. It's OK to use these words as placeholders or variable names in content for a technical audience, but if you can, use another placeholder instead, such as Example.exe."